sympy/solvers/ode/tests/test_ode.py

```diff
@@ -1446,124 +1446,6 @@ def test_1st_exact2_broken_check():
-    assert dsolve(eq6, hint='separable') == sol6
-    assert checkodesol(eq1, sol1, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq2, sol2, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq3, sol3, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq4, sol4, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq5, sol5, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq6, sol6, order=1, solve_for_func=False)[0]
-
-
-@slow
-def test_separable2():
-    a = Symbol('a')
-    eq6 = f(x)*x**2*f(x).diff(x) - f(x)**3 - 2*x**2*f(x).diff(x)
-    eq7 = f(x)**2 - 1 - (2*f(x) + x*f(x))*f(x).diff(x)
-    eq8 = x*log(x)*f(x).diff(x) + sqrt(1 + f(x)**2)
-    eq9 = exp(x + 1)*tan(f(x)) + cos(f(x))*f(x).diff(x)
-    eq10 = (x*cos(f(x)) + x**2*sin(f(x))*f(x).diff(x) -
-            a**2*sin(f(x))*f(x).diff(x))
-    sol6 = Eq(Integral((u - 2)/u**3, (u, f(x))),
-        C1 + Integral(x**(-2), x))
-    sol7 = Eq(-log(-1 + f(x)**2)/2, C1 - log(2 + x))
-    sol8 = Eq(asinh(f(x)), C1 - log(log(x)))
-    # integrate cannot handle the integral on the lhs (cos/tan)
-    sol9 = Eq(Integral(cos(u)/tan(u), (u, f(x))),
-        C1 + Integral(-exp(1)*exp(x), x))
-    sol10 = Eq(-log(cos(f(x))), C1 - log(- a**2 + x**2)/2)
-    assert dsolve(eq6, hint='separable_Integral').dummy_eq(sol6)
-    assert dsolve(eq7, hint='separable', simplify=False) == sol7
-    assert dsolve(eq8, hint='separable', simplify=False) == sol8
-    assert dsolve(eq9, hint='separable_Integral').dummy_eq(sol9)
-    assert dsolve(eq10, hint='separable', simplify=False) == sol10
-    assert checkodesol(eq6, sol6, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq7, sol7, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq8, sol8, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq9, sol9, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq10, sol10, order=1, solve_for_func=False)[0]
-
-
-def test_separable3():
-    eq11 = f(x).diff(x) - f(x)*tan(x)
-    eq12 = (x - 1)*cos(f(x))*f(x).diff(x) - 2*x*sin(f(x))
-    eq13 = f(x).diff(x) - f(x)*log(f(x))/tan(x)
-    sol11 = Eq(f(x), C1/cos(x))
-    sol12 = Eq(log(sin(f(x))), C1 + 2*x + 2*log(x - 1))
-    sol13 = Eq(log(log(f(x))), C1 + log(sin(x)))
-    assert dsolve(eq11, hint='separable') == sol11
-    assert dsolve(eq12, hint='separable', simplify=False) == sol12
-    assert dsolve(eq13, hint='separable', simplify=False) == sol13
-    assert checkodesol(eq11, sol11, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq12, sol12, order=1, solve_for_func=False)[0]
-    assert checkodesol(eq13, sol13, order=1, solve_for_func=False)[0]
-
-
-def test_separable4():
-    # This has a slow integral (1/((1 + y**2)*atan(y))), so we isolate it.
-    eq14 = x*f(x).diff(x) + (1 + f(x)**2)*atan(f(x))
-    sol14 = Eq(log(atan(f(x))), C1 - log(x))
-    assert dsolve(eq14, hint='separable', simplify=False) == sol14
-    assert checkodesol(eq14, sol14, order=1, solve_for_func=False)[0]
-
```

Comment:
this also returns False for `checkodesol()`

Reply:
Is the given solution correct?

Reply:
```
>>> eq=x*f(x).diff(x) + (1 + f(x)**2)*atan(f(x))
>>> dsolve(eq, hint='separable')
Eq(f(x), tan(C1/x))
>>> checkodesol(eq,_)
(False, -C1 + x*atan(tan(C1/x)))
```
I guess solution is correct i don't know why checkodesol returns false

Reply:
Probably for the same reasons as the other case. The solution is correct but the residual can't simplify without some assumptions on C1 and x.

This ODE has other solutions `f(x)=0` and `f(x)=I`. Can you note these other solutions and also add comments explaining why checkodesol doesn't work in these cases?

Reply:
can you verify that they both are failing for same reasons because i tried by declaring them as `real=True ` then too it didn't turn out be 0.

Reply:
> This ODE has other solutions `f(x)=0` and `f(x)=I`. Can you note these other solutions and also add comments explaining why checkodesol doesn't work in these cases?

yes, I have added these as comments so that in future we get a reference.

Reply:
You can see the residual here: `-C1 + x*atan(tan(C1/x))`. The problem is that `atan(tan(z))` only equals `z` for z in `(-pi/2, pi/2)`. This would simplify if `C1/x` was known to be in that range but it isn't because C1 and x are arbitrary complex symbols. Given an initial condition we could know the value of `C1` but the result would still only be a valid solution for some particular range of `x`.